spec/requests/api/v1/sharing/binary_request_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+describe Api::V1::Sharing::BinariesController do
+  let!(:application) { create :application }
+  let!(:user)        { create :user_with_various_binary_types }
+  let!(:token)       { create :access_token,
```

Comment:
I think we can get by with `let` instead of `let!` which would be preferred since it will only run when needed. `let!` will run before everything in the shared block.